src/transforms/coerce.jl

```diff
@@ -0,0 +1,35 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Coerce(pairs, tight=false, verbosity=1)
+
+Return a copy of the table, ensuring that the element scitypes of the columns match the new specification.
+
+This transform wraps around ScientificTypes.coerce function. Example syntax:
+
+table |> Coerce(:column1=>Continuous, :column2=>Count)
+"""
+struct Coerce{P} <: Transform
+  pairs::P
+  tight::Bool
+  verbosity::Int
+end
+
+Coerce(pair...; tight=false, verbosity=1) = Coerce(pair, tight, verbosity)
+
+isrevertible(::Type{<:Coerce}) = true
+
+function apply(transform::Coerce, table)
+  newtable = coerce(table, transform.pairs...;
+                    tight=transform.tight, verbosity=transform.verbosity)
+
+  scitypes = [elscitype(x) for x in Tables.columns(table)]
+  colnames = Tables.columnnames(table)
+  pairs = [Pair(i, j) for (i, j) in zip(colnames, scitypes)]
+  
+  newtable, pairs
+end
+
+revert(transform::Coerce, newtable, cache) = coerce(newtable, cache...)
```

Comment:
I think this is not strictly correct. Converting back to the original scitypes doesn't imply that the original machine types will be reverted. I think what we want here in the revert is a set of pairs mapping column names to machine types and then doing a traditional type cast in Julia to each column, using for example collect(T, x).